src/uct/ib/ud/base/ud_def.h

```diff
@@ -58,6 +58,7 @@ enum {
     UCT_UD_PACKET_FLAG_NAK     = UCS_BIT(27),
     UCT_UD_PACKET_FLAG_PUT     = UCS_BIT(28),
     UCT_UD_PACKET_FLAG_CTL     = UCS_BIT(29),
+    UCT_UD_PACKET_FLAG_CHECK   = UCS_BIT(30),
```

Comment:
no need for special flag, it should be just ACK+ACK_REQ

Reply:
i'm not sure that it would work (e.g. in case of user doesn't do AM/PUT operations), since we should increase local psn, and get ACKs on it

Reply:
yes, I checked, it doesn't work
`udx/test_ucp_peer_failure_keepalive.kill_receiver/0` doesn't detect an error, because `acked_psn == psn - 1` - so UD considers peer as alive.

Reply:
ic, need to psn++, so can it be a PUT message?

Reply:
> so can it be a PUT message?

do you mean to implement the same scheme as in RC? but it will consume receive WQ element on peer
I guess it was intent on Rc to not consume and repost receive WQ element